### ISSUE TITLE
Setting AppElement as aria-hidden="true" when modal is opened fails accessibility rules

### ISSUE THREAD

**Issue by petevliet** [reactjs/react-modal#752]
### Summary:

When the modal is opened, the site fails serious-level accessibility checks from axe's developer tooling. One could programmatically disable focusable elements on simple examples, but on a more complex site this is unfeasible.

### Steps to reproduce:

 1. Add Axe's Accessibility Checker tool as an [extension for Chrome](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US)
 2. open a react-modal and run a check against site
 3. Accessibility check fails for [this rule](https://dequeuniversity.com/rules/axe/3.2/aria-hidden-focus?application=AxeChrome)

### Expected behavior:

React-modal is promised to be Accessible, serious-level failures would preclude this designation.

### Link to example of issue:

Provided example linked from the Github documentation fails check for 'aria-hidden elements do not contain focusable elements' rule: [https://codepen.io/claydiffrient/pen/ENegGJ](https://codepen.io/claydiffrient/pen/ENegGJ)

**Comment by crswll** [comment]
This rule seems like it's meant to keep someone from accidentally doing this - not doing it on purpose. `react-modal` traps focus, so under normal circumstances you can't focus something inside the `AppElement` when the modal is open (essentially just like the screen reader would experience it), which I think is what they're worried about. I'm wondering if this is a false positive since it can't detect what you can focus and what you can't.

**Comment by diasbruno** [comment]
I think @crswll is correct about the false positive. The behavior described is correct.

As long as react-modal don't let the focus escape the modal, this should be fine.

**Comment by petevliet** [comment]
Thanks all, makes sense. I'll close this.
